### PR TITLE
refactor: use redux for track latest versions

### PR DESCRIPTION
**Pull request by Benjozork**
## Summary of Changes

* Use redux for storing state of latest mod track latest versions
* In a future PR: Use redux to store state of version history

Discord username (if different from GitHub): someperson#4953

### CODE REVIEW

File: src/renderer/components/AircraftSection/index.tsx [hunk 1]
```diff
@@ -55,15 +56,16 @@ type TransferredProps = {
...
+    installstatus: InstallStatus,
 }
 
 // All props
 type AircraftSectionProps = {
     mod: Mod,
     selectedtrack: ModTrack,
     installedtrack: ModTrack,
-    installstatus : InstallStatus,
+    installstatus : InstallStatus
```
Comment (FoxtrotSierra6829) [inline src/renderer/components/AircraftSection/index.tsx:67]:
```suggestion
    installstatus : InstallStatus,
```

File: src/renderer/components/App/index.tsx [hunk 2]
```diff
@@ -106,6 +109,21 @@ export const getModReleases = async (mod: Mod): Promise<ModVersion[]> => {
     return releases;
 };
 
+export const fetchLatestVersionNames = async (mod: Mod) => {
```
Comment (FoxtrotSierra6829) [inline src/renderer/components/App/index.tsx:112]:
```suggestion
export const fetchLatestVersionNames = async (mod: Mod): Promise<void>  => {
```

File: src/renderer/redux/reducers/latestVersionNames.reducer.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import * as actionTypes from '../actionTypes';
+import { SetModAndTrackLatestVersionName } from '../types';
+
+export type ModAndTrackLatestVersionNames = { modKey: string, trackKey: string, name: string }[]
```
Comment (FoxtrotSierra6829) [inline src/renderer/redux/reducers/latestVersionNames.reducer.ts:4]: All Redux types should be stored in `../types`

**Review by FoxtrotSierra6829 — CHANGES_REQUESTED** [review]

File: src/renderer/components/App/index.tsx [hunk 4]
```diff
@@ -213,6 +231,8 @@ function App() {
         }
     ];
 
+    mods.forEach(fetchLatestVersionNames);
```
Comment (nistei) [inline src/renderer/components/App/index.tsx:234]:
```suggestion
    useEffect(() => {
        mods.forEach(fetchLatestVersionNames);
    });
```

File: src/renderer/redux/reducers/latestVersionNames.reducer.ts (hunk 3 above)
Comment (nistei) [reply src/renderer/redux/reducers/latestVersionNames.reducer.ts:4]: Agreed. Imports aren't nice looking like this: `import { ModAndTrackLatestVersionNames } from "renderer/redux/reducers/latestVersionNames.reducer";`

File: src/renderer/components/AircraftSection/index.tsx [hunk 5]
```diff
@@ -55,15 +56,16 @@ type TransferredProps = {
...
+    latestVersionNames: ModAndTrackLatestVersionNames
 }
```
Comment (nistei) [inline src/renderer/components/AircraftSection/index.tsx:69]: Isn't `latestVersionNames` also coming from `connect()`? Should be moved to `ConnectedAircraftSectionProps`.
`AircraftSectionProps` should be an intersection type of `ConnectedAircraftSectionProps & TransferredProps` in that case.

**Review by nistei — CHANGES_REQUESTED** [review]

File: src/renderer/components/AircraftSection/index.tsx (hunk 5 above)
Comment (FoxtrotSierra6829) [reply src/renderer/components/AircraftSection/index.tsx:69]: talked with Holland about it converting to an intersection type before, but yes to reach this, additions should only go to connected or transferred. Agree on that

Comment (Benjozork) [reply src/renderer/components/AircraftSection/index.tsx:69]: will do

**Review by nistei — APPROVED** [review]
LGTM

**Pull request merged**